bundles/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/NetatmoBinding.java

```diff
@@ -52,732 +34,159 @@
-
-			final RefreshTokenResponse response = request.execute();
-			logger.debug("Response: {}", response);
-
-			if (response == null) {
-				throw new NetatmoException("Could not refresh access token! If you see "
-						+ "'Fatal transport error: javax.net.ssl.SSLHandshakeException' "
-						+ "above. You need to install the StartCom CA certificate and restart openHAB. "
-						+ "See https://github.com/openhab/openhab/wiki/Netatmo-Binding#missing-certificate-authority "
-						+ "for more information.");
-			}
-
-			this.accessToken = response.getAccessToken();
-
-			getStationsDataRequest = new GetStationsDataRequest(this.accessToken);
-			getStationsDataResponse = getStationsDataRequest.execute();
-		}
-
-	}
+public class NetatmoBinding extends AbstractActiveBinding<NetatmoBindingProvider>implements ManagedService {
+
+    private static final String DEFAULT_USER_ID = "DEFAULT_USER";
+
+    private static final Logger logger = LoggerFactory.getLogger(NetatmoBinding.class);
+
+    protected static final String CONFIG_CLIENT_ID = "clientid";
+    protected static final String CONFIG_CLIENT_SECRET = "clientsecret";
+    protected static final String CONFIG_REFRESH = "refresh";
+    protected static final String CONFIG_REFRESH_TOKEN = "refreshtoken";
+    protected static final String CONFIG_PRESSURE_UNIT = "pressureunit";
+    protected static final String CONFIG_UNIT_SYSTEM = "unitsystem";
+
+    /**
+     * The refresh interval which is used to poll values from the Netatmo server
+     * (optional, defaults to 300000ms)
+     */
+    private long refreshInterval = 300000;
+
+    private static Map<String, OAuthCredentials> credentialsCache = new HashMap<String, OAuthCredentials>();
+
+    private final NetatmoWelcomeBinding welcomeBinding = new NetatmoWelcomeBinding();
+    private final NetatmoWeatherBinding weatherBinding = new NetatmoWeatherBinding();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected String getName() {
+        return "Netatmo Refresh Service";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected long getRefreshInterval() {
+        return this.refreshInterval;
+    }
+
+    /**
+     * /**
+     * {@inheritDoc}
+     */
+    @SuppressWarnings("incomplete-switch")
+    @Override
+    protected void execute() {
+        logger.debug("Querying Netatmo API");
+        for (String userid : credentialsCache.keySet()) {
+
+            OAuthCredentials oauthCredentials = getOAuthCredentials(userid);
+            if (oauthCredentials.noAccessToken()) {
+                // initial run after a restart, so get an access token first
+                oauthCredentials.refreshAccessToken();
+            }
+
+            // Start the execution of the weather station and welcome binding
+            weatherBinding.execute(oauthCredentials, this.providers, this.eventPublisher);
+            welcomeBinding.execute(oauthCredentials, this.providers, this.eventPublisher);
```

Comment:
done, i check if wether a wether or camera device ist configured as item. If the items are not read at the first run, the execution is delayed till the next refresh interval